vlog4j-parser/src/test/java/org/semanticweb/vlog4j/syntax/parser/RuleParserTest.java

```diff
@@ -438,4 +454,82 @@ public void testCustomDatatype() throws ParsingException {
 		DatatypeConstant result = (DatatypeConstant) literal.getConstants().toArray()[0];
 		assertEquals(constant, result);
 	}
+
+	@Test
+	public void testGetFacts() throws ParsingException, IOException {
+		KnowledgeBase kb = new KnowledgeBase();
+		final DataSource dataSource1 = new SparqlQueryResultDataSource(new URL("https://example.org/"), "var",
+				"?var wdt:P31 wd:Q5 .");
+		final Predicate predicate1 = Expressions.makePredicate("country", 1);
+		final DataSourceDeclaration dataSourceDeclaration1 = new DataSourceDeclarationImpl(predicate1, dataSource1);
+		final DataSource dataSource2 = new SparqlQueryResultDataSource(new URL("https://example.org/"), "var",
+				"?var wdt:P31 wd:Q5 .");
+		final Predicate predicate2 = Expressions.makePredicate("inEuropeOutsideGermany", 1);
+		final DataSourceDeclaration dataSourceDeclaration2 = new DataSourceDeclarationImpl(predicate2, dataSource2);
+		final DataSource dataSource3 = new SparqlQueryResultDataSource(new URL("https://example.org/"), "var2",
+				"?var2 wdt:P31 wd:Q5 .");
+		final DataSourceDeclaration dataSourceDeclaration3 = new DataSourceDeclarationImpl(predicate2, dataSource3);
+		final Predicate predicate4 = Expressions.makePredicate("city", 1);
+		final DataSourceDeclaration dataSourceDeclaration4 = new DataSourceDeclarationImpl(predicate4, dataSource2);
+		RuleParser.parseInto(kb, fact.toString() + ".");
+		RuleParser.parseInto(kb, dataSourceDeclaration1.toString());
+		RuleParser.parseInto(kb, dataSourceDeclaration2.toString());
+		RuleParser.parseInto(kb, dataSourceDeclaration3.toString());
+		RuleParser.parseInto(kb, dataSourceDeclaration4.toString());
```

Comment:
you can just add these Statements (fact, dataSourceDeclaration1...4) to the kb, with addStatement, you don't need to call toString on them, and then parse them into equivalent objects, and them add those equiv statements to the kb.